extensions/ql-vscode/src/query-history/query-history-scrubber.ts

```diff
@@ -132,46 +132,53 @@ async function scrubDirectory(
   errorMsg?: string;
   deleted: boolean;
 }> {
-  const timestampFile = join(dir, "timestamp");
   try {
-    let deleted = true;
-    if (!(await stat(dir)).isDirectory()) {
-      void extLogger.log(`  ${dir} is not a directory. Deleting.`);
-      await remove(dir);
-    } else if (!(await pathExists(timestampFile))) {
-      void extLogger.log(`  ${dir} has no timestamp file. Deleting.`);
-      await remove(dir);
-    } else if (!(await stat(timestampFile)).isFile()) {
-      void extLogger.log(`  ${timestampFile} is not a file. Deleting.`);
+    if (await shouldScrubDirectory(dir, now, maxQueryTime)) {
       await remove(dir);
+      return { deleted: true };
     } else {
-      const timestampText = await readFile(timestampFile, "utf8");
-      const timestamp = parseInt(timestampText, 10);
-
-      if (Number.isNaN(timestamp)) {
-        void extLogger.log(
-          `  ${dir} has invalid timestamp '${timestampText}'. Deleting.`,
-        );
-        await remove(dir);
-      } else if (now - timestamp > maxQueryTime) {
-        void extLogger.log(
-          `  ${dir} is older than ${maxQueryTime / 1000} seconds. Deleting.`,
-        );
-        await remove(dir);
-      } else {
-        void extLogger.log(
-          `  ${dir} is not older than ${maxQueryTime / 1000} seconds. Keeping.`,
-        );
-        deleted = false;
-      }
+      return { deleted: false };
     }
-    return {
-      deleted,
-    };
   } catch (err) {
     return {
       errorMsg: `  Could not delete '${dir}': ${err}`,
       deleted: false,
     };
   }
 }
+
+async function shouldScrubDirectory(
+  dir: string,
+  now: number,
+  maxQueryTime: number,
+): Promise<boolean> {
+  const timestamp = await getTimestamp(join(dir, "timestamp"));
+  if (timestamp === undefined || Number.isNaN(timestamp)) {
+    void extLogger.log(`  ${dir} timestamp is missing or invalid. Deleting.`);
+    return true;
+  } else if (now - timestamp > maxQueryTime) {
+    void extLogger.log(
+      `  ${dir} is older than ${maxQueryTime / 1000} seconds. Deleting.`,
+    );
+    return true;
+  } else {
+    void extLogger.log(
+      `  ${dir} is not older than ${maxQueryTime / 1000} seconds. Keeping.`,
+    );
+    return false;
+  }
+}
+
+async function getTimestamp(
+  timestampFile: string,
+): Promise<number | undefined> {
+  try {
+    const timestampText = await readFile(timestampFile, "utf8");
+    return parseInt(timestampText, 10);
+  } catch (err) {
+    void extLogger.log(
+      `  Could not read timestamp file '${timestampFile}': ${err}`,
```

Comment:
Should we be using `getErrorMessage` here?

Reply:
Good point, we should do. I was being a bit lazy here by just copying a different log call from this file.